shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionActionDispatcher.kt

```diff
@@ -111,17 +114,27 @@ class StepCompletionActionDispatcher(
                 if (topicProgress.isCompleted) {
                     topicCompletedFlow.notifyDataChanged(action.topicId)
 
-                    val topicTitle = topicsInteractor
-                        .getTopic(action.topicId)
+                    val topicTitleResult = actionScope.async {
+                        topicsInteractor.getTopic(action.topicId)
+                    }
+                    val nextStepIdResult = actionScope.async {
+                        topicsToDiscoverNextInteractor.getNextTopicToDiscover()
+                    }
```

Comment:
It's better to wrap the whole `Action.CheckTopicCompletionStatus` handling with `coroutineScope` instead of crating new coroutine hierarchy from `actionScope`.
See  docs
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/coroutine-scope.html
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/supervisor-scope.html